Nostalgia/InRoomChat.cs

```diff
@@ -26,101 +27,148 @@ public void AddLine(string newLine)
 
     public void OnGUI()
     {
-        if (this.IsVisible && (PhotonNetwork.connectionStateDetailed == global::PeerState.Joined))
+        if (!this.IsVisible && (PhotonNetwork.connectionStateDetailed != global::PeerState.Joined))
         {
-            if ((Event.current.type == EventType.KeyDown) && ((Event.current.keyCode == KeyCode.KeypadEnter) || (Event.current.keyCode == KeyCode.Return)))
+            return;
+        }
+        if ((Event.current.type == EventType.KeyDown) && ((Event.current.keyCode == KeyCode.KeypadEnter) || (Event.current.keyCode == KeyCode.Return)))
+        {
+            if (!string.IsNullOrEmpty(this.inputLine))
             {
-                if (!string.IsNullOrEmpty(this.inputLine))
+                /**
+                 * This regex is used to parse input chat commands
+                 * @command and @args are used in "/command arg" type input
+                 * @single is used when "/command" type input
+                 * @tab because tab
+                 * @any when any normal chat input
+                 */
+                Match matches = Regex.Match(this.inputLine, @"\\|\/(?<command>\w*)\s(?<args>[^\s].*)|\\|\/(?<single>\w*)|(?<tab>\t)|(?<any>^[^\\\/].*?$)");
```

Comment:
Regex is wrong:
- `\\|\/` need to be in parentheses (non-capturing) or the `OR` will be applied to the whole regex. Use `[\\\/]` instead.
- The `args` capturing group is wrong. It will match a non-space followed by any number of characters.
- Make args optional and remove the single catching group.
- I'd suggest using a regex to match commands. If it does not match then send the message to chat.

Example:
`[\\\/]([^\s]+)(?:\s+(.*))?`
This regex will always match every command (slash followed by anything but a space is considered a command). Group 1 will be the command name (use `.ToUpperInvariant()` or `Equals(..;, StringComparison.OrdinalIgnoreCase)` to check if command exists and execute it) while Group 2 will be the args or `string.Empty` if none were supplied.
Then you can use Match.Success to check if the input string was recognized as command and if not you send the string to chat.

Reply:
Thanks! I will apply that.